cmd/admin-storage-class-add.go

```diff
@@ -0,0 +1,296 @@
+		}
+
+		bucket := ctx.String("bucket")
+		if bucket == "" {
+			fatalIf(errInvalidArgument().Trace(), fmt.Sprintf("%s backend requires target bucket", scType))
+		}
+
+		s3Opts := []madmin.S3Options{}
+		prefix := ctx.String("prefix")
+		if prefix != "" {
+			s3Opts = append(s3Opts, madmin.S3Prefix(prefix))
+		}
+
+		endpoint := ctx.String("endpoint")
+		if endpoint != "" {
+			s3Opts = append(s3Opts, madmin.S3Endpoint(endpoint))
+		}
+
+		region := ctx.String("region")
+		if region != "" {
+			s3Opts = append(s3Opts, madmin.S3Region(region))
+		}
+
+		s3SC := ctx.String("storage-class")
+		if s3SC != "" {
+			s3Opts = append(s3Opts, madmin.S3StorageClass(s3SC))
+		}
+
+		s3Cfg, err := madmin.NewTransitionStorageClassS3(scName, accessKey, secretKey, bucket, s3Opts...)
+		if err != nil {
+			fatalIf(probe.NewError(err), "Invalid configuration for s3 backed storage-class")
+		}
+
+		return madmin.StorageClassConfig{
+			Type: madmin.S3,
+			S3:   s3Cfg,
+		}
+	case madmin.Azure:
+		accessKey := ctx.String("access-key")
+		secretKey := ctx.String("secret-key")
+		if accessKey == "" || secretKey == "" {
+			fatalIf(errInvalidArgument().Trace(), fmt.Sprintf("%s backend requires access credentials", scType))
+		}
+
+		bucket := ctx.String("bucket")
+		if bucket == "" {
+			fatalIf(errInvalidArgument().Trace(), fmt.Sprintf("%s backend requires target bucket", scType))
+		}
+
+		azOpts := []madmin.AzureOptions{}
+		endpoint := ctx.String("endpoint")
+		if endpoint != "" {
+			azOpts = append(azOpts, madmin.AzureEndpoint(endpoint))
+		}
+
+		prefix := ctx.String("prefix")
+		if prefix != "" {
+			azOpts = append(azOpts, madmin.AzurePrefix(prefix))
+		}
+
+		azCfg, err := madmin.NewTransitionStorageClassAzure(scName, accessKey, secretKey, bucket, azOpts...)
+		if err != nil {
+			fatalIf(probe.NewError(err), "Invalid configuration for azure backed storage-class")
+		}
+
+		return madmin.StorageClassConfig{
+			Type:  madmin.Azure,
+			Azure: azCfg,
+		}
+	case madmin.Gcs:
+		bucket := ctx.String("bucket")
+		if bucket == "" {
+			fatalIf(errInvalidArgument().Trace(), fmt.Sprintf("%s backend requires target bucket", scType))
+		}
+
+		gcsOpts := []madmin.GcsOptions{}
+		prefix := ctx.String("prefix")
+		if prefix != "" {
+			gcsOpts = append(gcsOpts, madmin.GcsPrefix(prefix))
+		}
+
+		region := ctx.String("region")
+		if region != "" {
+			gcsOpts = append(gcsOpts, madmin.GcsRegion(region))
+		}
+
+		credsPath := ctx.String("credentials-file")
+		credsBytes, err := ioutil.ReadFile(credsPath)
+		if err != nil {
+			fatalIf(probe.NewError(err), "Failed to read credentials file")
+		}
+
+		gcsCfg, err := madmin.NewTransitionStorageClassGcs(scName, credsBytes, bucket, gcsOpts...)
+		if err != nil {
+			fatalIf(probe.NewError(err), "Invalid configuration for gcs backed storage-class")
+		}
+
+		return madmin.StorageClassConfig{
+			Type: madmin.Gcs,
+			Gcs:  gcsCfg,
+		}
+	}
+	return madmin.StorageClassConfig{}
```

Comment:
the command line should also take the target storage class where object should be tiered to. Seems to be missing for all 3 clouds